src/scripts/create_icelandic_qa.py

```diff
@@ -0,0 +1,156 @@
+
+        Args:
+            question:
+                The question.
+            answer:
+                The answer.
+            context:
+                The context.
+
+        Returns:
+            The rephrased answer (if the answer is already in the context, it is returned as is).
+        """
+        answer = answer[:-1] if answer.endswith(".") else answer
+
+        if answer.lower() in context.lower():
+            return answer
+
+        # Use OpenAI to rephrase the answer
+        chat_completion = client.chat.completions.create(
+            messages=[
+                {
+                    "role": "user",
+                    "content": (
+                        f"Given the following context: '{context}', please rephrase the answer '{answer}' so that it matches exactly "
+                        f"with a phrase from the context in a case-insensitive way. E.g. it must be possible to find the rephrased answer "
+                        f"in the context without any modifications. "
+                        f"The rephrased answer should be as concise as possible, preferable not more than 7 words, and ideally 3 or less words."
+                        f"Here is the related question: '{question}'"
+                    ),
+                }
+            ],
+            model="gpt-4o",
+        )
+
+        rephrased_answer = chat_completion.choices[0].message.content
+        return rephrased_answer
+
+    # Rephrase the answers
+    df_with_no_outliers["answer"] = df_with_no_outliers.apply(
+        lambda row: rephrase_answer(row["question"], row["answer"], row["context"]),
+        axis=1,
+    )
+
+    # Only work with the questions having answers in the context
+    has_answer: pd.Series = df_with_no_outliers.apply(
+        lambda row: row["answer"].lower() in row["context"].lower(), axis=1
+    )
+
+    df_with_answer: pd.DataFrame = df_with_no_outliers.loc[has_answer]
+
+    # Make the `answers` column a dictionary with the answer and the answer start
+    df_with_answer["answers"] = df_with_answer.apply(
+        lambda row: {
+            "text": [row["answer"]],
+            "answer_start": row["context"].lower().index(row["answer"].lower()),
+        },
+        axis=1,
+    )
+
+    # Remove the old `answer` column
+    df_with_answer = df_with_answer.drop(columns=["answer"])
+
+    # Create validation split
+    val_size = 128
+    val_df = df_with_answer.sample(n=val_size, random_state=4242)
+
+    # Create test split
+    test_size = 1024
+    df_with_answer_filtered: pd.DataFrame = df_with_answer.loc[
+        ~df_with_answer.index.isin(val_df.index)
+    ]
+    test_df = df_with_answer_filtered.sample(n=test_size, random_state=4242)
+
+    # Create train split
+    train_size = 531
+    full_train_df_with_answer = df_with_answer_filtered.loc[
+        ~df_with_answer_filtered.index.isin(test_df.index)
+    ]
+    train_df = full_train_df_with_answer.sample(n=train_size, random_state=4242)
```

Comment:
It would be a bit more robust if we simply take the remaining samples here, rather than specifically 531:

```suggestion
    # Create train split
    train_df = df_with_answer_filtered.loc[
        ~df_with_answer_filtered.index.isin(test_df.index)
    ]
    
    # Sanity check that we include enough training samples
    train_num_samples_lower_bound = 500
    assert len(train_df) > train_num_samples_lower_bound, (
        f"There are only {len(train_df):,} samples in the training set - there should be at "
        f"least {train_num_samples_lower_bound:,}."
    )
```